src/scriptdev2/scripts/kalimdor/durotar.cpp

```diff
@@ -130,11 +225,12 @@ bool EffectDummyCreature_lazy_peon_awake(Unit* pCaster, uint32 uiSpellId, SpellE
     // always check spellid and effectindex
     if (uiSpellId == SPELL_AWAKEN_PEON && uiEffIndex == EFFECT_INDEX_0)
     {
+        //TODO some of these checks are now redundant, should be removed by someone who knows how to do so without breaking things
```

Comment:
Doesn't exactly answer what true/false do, i.e. what is whatever is calling the dummy check thing doing with the return value. But it looks like it basically should just return true in the middle part (and awake the peon obviously). Not sure if it's even possible for uiSpellId to be wrong, so maybe that check is redundant too.